src/lib.rs

```diff
@@ -915,6 +915,45 @@ fn lexicographical_topological_sort(
     Ok(PyList::new(py, out_list).into())
 }
 
+#[pyfunction]
+fn graph_greedy_color(
+    py: Python,
+    graph: &graph::PyGraph,
+) -> PyResult<PyObject> {
+    let mut colors: HashMap<usize, usize> = HashMap::new();
+    let mut node_vec: Vec<NodeIndex> = graph.graph.node_indices().collect();
+    let node_count = graph.graph.node_count();
+    let mut sort_map: HashMap<NodeIndex, usize> = HashMap::new();
+    for k in node_vec.iter() {
+        sort_map.insert(*k, node_count - graph.graph.edges(*k).count());
```

Comment:
It might be good to add a comment "It is intended to sort nodes in the descending order of degree" because `node_count - graph.graph.edges(*k).count()` does not look intuitive.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/77/commits/b63f6bc0bf30ba3e5e51cf92a3df4dfb01cf20f5